core/src/commonMain/kotlin/State.kt

```diff
@@ -108,3 +110,34 @@ public sealed class State {
         }
     }
 }
+
+/**
+ * Returns `true` if `this` is at least in state [T], where [State]s are ordered:
+ * - [State.Disconnected] (smallest)
+ * - [State.Disconnecting]
+ * - [State.Connecting.Bluetooth]
+ * - [State.Connecting.Services]
+ * - [State.Connecting.Observes]
+ * - [State.Connected] (largest)
+ */
+internal inline fun <reified T: State> State.isAtLeast(): Boolean {
+    val currentState = when (this) {
+        is State.Disconnected -> 0
+        State.Disconnecting -> 1
+        State.Connecting.Bluetooth -> 2
+        State.Connecting.Services -> 3
+        State.Connecting.Observes -> 4
+        State.Connected -> 5
+        else -> error("Unreachable.")
```

Comment:
With `State` being a sealed class, isn't this exhaustive without the `else`?

Reply:
I believe so, but it refused to compile (and I'm not sure why).

Reply:
I guess the compiler still can't handle nested sealed classes. Human still smarter than computer...I'll chalk that up as a win

Reply:
Nevermind. The IDE is highlighting is at an error but it actually does compile. Thanks for making me double check.

Reply:
Human smarter than IDE...still win?